teps/0007-conditions-beta.md

```diff
@@ -0,0 +1,499 @@
+spec:
+  workspaces:
+    - name: source-repo
+  params:
+    - name: path
+      default: 'README.md'
+  tasks: 
+    - name: create-file
+      taskRef: 
+        name: create-readme-file
+      workspaces:
+        - name: source
+          workspace: source-repo
+    - name: file-exists
+      taskRef: 
+        name: file-exists
+      workspaces:
+        - name: source
+          workspace: source-repo
+    - name: echo-file-exists 
+      when:
+        - name: check-file-exists
+          input: '$(tasks.file-exists.results.exists)'
+          operator: In
+          values: ['true']
+      taskRef:
+        name: echo-file-exists
+```
+
+Other examples of `Guards` using `Match Expressions` are:
+
+```yaml
+name: branch-is-main
+input: $(params.branch)
+operator: In
+values: [‘main’]
+---
+name: branch-not-main
+input: $(params.branch)
+operator: NotIn
+values: [‘main’]
+---
+name: always-false
+input: ‘false’
+operator: In
+values: [‘’]
+```
+
+We can explore adding more `Operators` later if needed, such as `IsTrue`, `IsFalse`, `IsEmpty` and `IsNotEmpty`.
+
+### Skipping
+
+Provide more flexibility when a `Guard` executes as `False` by adding a field, `continueAfterSkip`, used to specify whether execute the `Tasks` that are ordering-dependent on the skipped guarded `Task`. The `continueAfterSkip` field defaults to `false`/`no` and users can set it to `true`/`yes` (case insensitive) to allow for execution of the rest of the branch. 
+
+A `Task` branch is made up of dependent `Tasks`, where there are two types of dependencies:
+- _Resource dependency_: based on resources needed from parent `Task`, which includes Workspaces, `Results` and Resources. 
+- _Ordering dependency_: based on runAfter which provides sequencing of `Tasks` when there may not be resource dependencies. 
+
+Setting `continueAfterSkip` on a guarded `Task` with ordering dependencies is valid and the subsequent `Tasks` should execute. However, setting `continueAfterSkip` on a guarded `Task` with resource dependencies is invalid because those dependent `Tasks` will have resource validation errors and fail the whole `Pipeline`. We will add validation to confirm the dependencies allow for passing in the `continueAfterSkip` field. 
+
+```yaml
+api: tekton.dev/v1alpha1
+kind: Pipeline
+metadata: 
+    name: generate-file
+spec:
+  workspaces:
+    - name: source-repo
+  params:
+    - name: path
+      default: 'README.md'
+  tasks: 
+    - name: create-file # executed
+      taskRef: 
+        name: create-readme-file
+      workspaces:
+        - name: source
+          workspace: source-repo
+    - name: file-exists # executed
+      taskRef: 
+        name: file-exists
+      workspaces:
+        - name: source
+          workspace: source-repo
+    - name: echo-file-does-not-exist # skipped 
+      when:
+        - name: check-file-exists
+          input: '$(tasks.file-exists.results.exists)'
+          operator: In
+          values: ['false']
+      taskRef:
```

Comment:
Is it supposed to be a `continueAfterSkip: "true"` field here?

Reply:
that's right, must have mistakenly removed it while adding workspaces